src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx

```diff
@@ -9,26 +21,79 @@ import { Empty, List } from "antd";
  * @constructor
  */
 export function SampleMetadata({ metadata }) {
-  return Object.keys(metadata).length ? (
-    <List
-      itemLayout="horizontal"
-      dataSource={Object.keys(metadata).sort((a, b) => a.localeCompare(b))}
-      renderItem={(item) => (
-        <List.Item className="t-sample-details-metadata-item">
-          <List.Item.Meta
-            title={
-              <span className="t-sample-details-metadata__field">{item}</span>
-            }
-            description={
-              <span className="t-sample-details-metadata__entry">
-                {metadata[item].value}
-              </span>
-            }
+  const [visible, setVisible] = React.useState(false);
+  return (
+    <>
+      <Button onClick={() => setVisible(true)}>
+        {i18n("SampleMetadata.addNewMetadata")}
+      </Button>
+      <Modal
+        className="t-add-metadata-field"
+        onCancel={() => setVisible(false)}
+        visible={visible}
+        onOk={() => setVisible(false)}
+      >
+        <Title level={4}>{i18n("SampleMetadata.modal.title")}</Title>
+        <Form>
+          <Form.Item
+            name="metadata_field_name"
+            label={i18n("SampleMetadata.modal.fieldName")}
+          >
+            <Input />
+          </Form.Item>
+          <Form.Item
+            name="metadata_field_value"
+            label={i18n("SampleMetadata.modal.fieldValue")}
+          >
+            <Input />
+          </Form.Item>
+          <Form.Item
+            name="metadata_field_permission"
+            label={i18n("SampleMetadata.modal.permission")}
+          >
+            <Select
+              defaultValue="1"
+              style={{ width: "100%" }}
+              onChange={(e) =>
+                console.log("Changed to LEVEL " + e + " metadata permission.")
+              }
+            >
+              <Option value="1">{i18n("SampleMetadata.modal.level1")}</Option>
```

Comment:
Do you think we should make getting these an ajax request?  That way if they change we only need to update the server and not different places in the UI?

Reply:
Yup that's the plan :)